### PR TITLE
Dev UI Documentation

### LINKED ISSUE #32597: Dev UI Documentation
Update and enhance the Documentation to reflect the new Dev UI

### PR DESCRIPTION
**Pull request by phillip-kruger**
Fix #32597

This PR is adding / updating the Dev UI Guide for the new Dev UI. 

It's still a WIP, but in case anyone wants to review what is there in the mean time

### CODE REVIEW

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 1]
```diff
@@ -0,0 +1,244 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Dev UI
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:6]: That's typically a guide which should follow the diatixys template - I would say reference guide.

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+image::dev-ui-overview-v2.png[alt=Dev UI overview,role="center"]
+
+It allows you to quickly visualize all the extensions currently loaded, see their status and go directly
+to their documentation. You can also see and change Configuration, manage and visualize Continuous Testing, see Dev Services information, See the Build information, and also see and stream various logs.
+
+Each extension used in the application will be listed and you can navigate to the guide for each extension, see some more information on the extension, and view configuration applicable for that extension:
+
+image::dev-ui-extension-card-v2.png[alt=Dev UI extension card,role="center"]
+
+== How can I make my extension support the Dev UI?
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:32]: I would rephrase it to `Make my extension extend the Dev UI`

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+Extension can:
+
+- <<add-links-to-their-card,Add links to their card>> 
+- <<add-full-pages,Add full custom pages>>
+- <<add-a-log-file,Add a log stream>>
+- <<add-a-sub-menu,Add a sub-menu>>
+- <<custom-cards,Custom cards>>
+
+== Add links to their card
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:46]: Add links to an extension card

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+There are a few options to add a link to a card:
+
+=== External Links
+
+These are links that reference other (external from Dev UI) data. This data can be HTML pages or text/pfd/other data.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:52]: pdf, no?

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+----
+CardPageBuildItem pageBuildItem = new CardPageBuildItem();
+pageBuildItem.addBuildTimeData("someKey", getSomeValueObject());
+----
+
+You can add mulitple of these key-value pair for all the data you know at build time that you need on the page.
+
+There are a few options to add full page content in Dev UI. Starting from the most basic (good start) to a full blown web-component (preferred).
+
+=== Just display some build time data on a screen (without having to do frontend coding):
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:139]: Remote just

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+If you have some data that is known at build time that you want to display you can use one of the following builders in `Page`:
+
+==== 1. Raw data
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:143]: Avoid numeration here.

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+                .buildTimeDataKey("someKey")); // <2>
+----
+<1> Use the `rawDataPageBuilder`.
+<2> Link back to the key used when you added the build time data in `addBuildTimeData` on the Page BuildItem.
+
+That will create a link to a page that renders the data in json:
+
+image::dev-ui-raw-page-v2.png[alt=Dev UI raw page,role="center"]
+
+==== 2. Table data
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:159]: Avoid numeration

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+----
+<1> Use the `tableDataPageBuilder`.
+<2> Optionally only show certain fields.
+<3> Link back to the key used when you added the build time data in `addBuildTimeData` on the Page BuildItem.
+
+That will create a link to a page that renders the data in a table:
+
+image::dev-ui-table-page-v2.png[alt=Dev UI table page,role="center"]
+
+==== 3. Qute data
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:180]: Avoid numeration

File: docs/src/main/asciidoc/dev-ui-v2.adoc [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+You can also display your Build time data using a qute template. All build time data keys is available to use int the template:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/dev-ui-v2.adoc:182]: why uppercase B?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/dev-ui-v2.adoc (hunk 1 above)
Comment (phillip-kruger) [reply docs/src/main/asciidoc/dev-ui-v2.adoc:6]: I copied from the old dev ui documentation. What is dialixys and where can I lean about how this should look ?

**Review by cescoffier — APPROVED** [review]
